src/action.cpp

```diff
@@ -520,10 +520,6 @@ point get_delta_from_movement_direction( action_id act )
     }
 }
 
-// Get the key for an action, used in the action menu to give each action the
-// hotkey it is bound to.
-// We ignore bindings to '?' because that will already do something else in
-// this menu (open the menu keybindings).
 long hotkey_for_action( action_id action )
```

Comment:
This doesn't return values properly for function keys, e.g. if debug action is set for F10, `hotkey_for_action` would return -1, thus rendering debug menu unavailable (tested on Windows with VS and MSYS).

Reply:
Yep you're right, I can confirm this behavior, functions keys are not understood by `keys_bound_to`. Hmmm I don't know exactly what you should be the next step here...

Reply:
This is the check that returns the bound key from `keys_bound_to` (which returns nothing at the moment):

https://github.com/CleverRaven/Cataclysm-DDA/blob/f2df59c59dc1b3f3d8350c758ca041c903e797bd/src/input.cpp#L667-L670
The above comparison with <kbd>F9</kbd> bound to the debug menu:

* `events_event.type == CATA_INPUT_KEYBOARD`: true 
* `events_event.sequence.size() == 1`: true
* `events_event.sequence.front() < 0xFF`: false (0x111 for F9 for ex.)
* `isprint( events_event.sequence.front()`: false (0x111 is not printable)

So technically there's a bound key but the check enforces that it must be a 8-bit key code and be printable...

Is it worth a new function or a patch? should I kill this PR?

Reply:
We can address this in another PR. It is not your fault and this function behavior existed before.

Reply:
Ok understood! Thanks for your feedback. I'll try to address this in another PR as you suggested. Thank you!